pythran/config.py

```diff
@@ -246,18 +245,24 @@ def parse_define(define):
         extension['define_macros'].append('PYTHRAN_BLAS_NONE')
 
     if user_blas not in reserved_blas_entries:
-        # Numpy can pollute stdout with checks
-        with silent():
-            numpy_blas = numpy_sys.get_info(user_blas)
-            # required to cope with atlas missing extern "C"
-            extension['define_macros'].append('PYTHRAN_BLAS_{}'
-                                              .format(user_blas.upper()))
-            extension['libraries'].extend(numpy_blas.get('libraries', []))
-            extension['library_dirs'].extend(
-                numpy_blas.get('library_dirs', []))
-            extension['include_dirs'].extend(
-                numpy_blas.get('include_dirs', []))
-
+        if sys.version_info < (3, 12):
+            # `numpy.distutils` not present for Python >= 3.12
+            try:
+                import numpy.distutils.system_info as numpy_sys
+                 # Numpy can pollute stdout with checks
+                with silent():
+                    numpy_blas = numpy_sys.get_info(user_blas)
+                    # required to cope with atlas missing extern "C"
+                    extension['define_macros'].append('PYTHRAN_BLAS_{}'
+                                                      .format(user_blas.upper()))
+                    extension['libraries'].extend(numpy_blas.get('libraries', []))
+                    extension['library_dirs'].extend(
+                        numpy_blas.get('library_dirs', []))
+                    extension['include_dirs'].extend(
+                        numpy_blas.get('include_dirs', []))
+            except Exception:
+                # This may fail, because of setuptools backwards compat issues
```

Comment:
In that case, it should probably be an hard error, or at least a warning

Reply:
I agree. After some thinking about it, this is only opt-in functionality that isn't used by SciPy - and probably used very rarely. If it does not work, better for the user to know that.